src/components/App.js

```diff
@@ -1,39 +1,55 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
-import { checkURIforError, getTokenFromURI, getAuthURL } from '../config/authConfig';
-import macaroon from '../assets/img/Macaroonicon.png';
+import { checkURIforError, getTokenFromURI, getTokenFromLocal } from '../config/authConfig';
+import { Dashboard } from './Dashboard';
+import { Sidebar } from './Sidebar';
+import { Welcome } from './Welcome';
+import { LoginLink } from './LoginLink';
+import UserPlaylists from './UserPlaylists';
+import CurrentPlaylist from './CurrentPlaylist';
 
 
 class App extends React.Component {
   state = {
     token: '',
-    user: '',
-    error: false,
-    errorMessage: '',
+    user: {
+      name: '',
+      id: '',
+      error: false,
+      errorMessage: '',
+    },
+    currentPlaylist: {
+      name: '',
+    },
   }
 
   componentDidMount() {
+    let token = getTokenFromURI();
     const authError = checkURIforError();
-    const token = getTokenFromURI();
+    if (!token) {
+      token = getTokenFromLocal();

```

Comment:
I just threw this together token persistence functionality in localStorage so I'd stay logged in throughout refreshes, not meant for production in its current form.